packages/bs-flopflip/src/reactBroadcast.re

```diff
@@ -0,0 +1,131 @@
+type any;
+
+/* Since in ReasonReact we can't pass a component module as a prop (*),
+   we need to pass it as a function, which is still a React component.
+   https://reasonml.github.io/reason-react/docs/en/component-as-prop.html
+
+   (*) well, we kind of can 😅
+   [from the Discord channel:
+     "Sure you can, just give it a signature"
+     https://astrada.github.io/reason-react-playground/?reason=LYewJgrgNgpgBAFwJ4Ad4CUQQfAvHAbwCg45YE5gBDAaxgC44AKAPxSoQAt6BnBAJwCWAOwDmAGjhV+-KkiboYVHiGGKqAYwQA6fkq0BRWMBjCEASnNxcAPjjqVa-To0hgKVaYQKlj9Vu0+DhhYHh5JB1V-HWEQRQQqERgwAAV+EBRw+18o521NBEFVUJ5zAG4iAF8KolBIWHssHH5rQhIyGApXd08zVsinAsCEnBKAYTcPYS8mACJMbBh+WfL28kpaPGY9YTAlxiY66AwmmCtbbOVcob0CoxgTM3O7YlJSbQ-uqa9xdtI-uA7Pb8RgAfR4IQAZtYXgDSOt0ot+ABlEZbWbAJCo4KzCpvN5wyjgY6NRatV74t7rL69Cj4AbRYbBcaTWlzBY4FZ4ynwzobOitVjsLiSUEaTiCKBgHbPNo8-EfbQ06ZmX7ygnq0hA-ZwcFQmGE-EASAAPGBBAA3GyGt5Go0+K6DAJ8IRiAAqIHuj28iOa2JwcAA1IG4MLOJYbaRTQB6c1Wm2VQnVQmE7X8Q7Ehoc+CMbOrfGJnntZPJojRAAiAHkALKKXSmYEer1eADqgi4AEkwEx2ibs-xrQqPkwR0cs6dLAaeaa44PKUbSH3TqGOJxcLNo1RZnBo3PjYvsyuuOvowAjbe7m0m2OWmzmXvR-s2SSzFB6C2CGAAdy5QA
+   ]
+   */
+type reactClassForComponentProp = unit => ReasonReact.reactElement;
+
+type flagName = string;
+
+type flagVariation;
```

Comment:
This could be `string | boolean` depending on how union types work?

Reply:
As far as I know it's discouraged to do that: https://reasonml.github.io/guide/language/variant/#variants-must-have-constructors

I haven't looked deeper to see what we can do in such cases.

Btw: if you just provide a type without a value, it's called an "abstract" type

>  It's a convenient way of specifying the relationship between external calls without knowing what the shape of the data is under the hood.

https://reasonml.github.io/guide/javascript/converting#clean-up-types-pass-2
https://reasonml.github.io/guide/language/module#signatures

Reply:
Interesting and it makes sense from a type soundness perspective. To me it‘s a bit of a leaky business logic thing here. Flag variations, in almost all platforms with multi variate flags I know, are bools or strings. Which in turn leaks into code I guess. I guess I can not refactor this in accordance with the ˋdrawLineˋ and ˋdrawStringˋ example. I could just have two type contructors even if discouraged or work with abstract types (which feel like generics). So a ˋFlagVariation<T>ˋ. The latter might make sense as I actually never have care what concrete type it is. As long as I can === it. 

Thanks for the thought provocing links. I like this stuff.

Reply:
Consider that this is just the interop interface so that the compiler can understand what's going on. In JS land is going to work as before (e.g. the `===` part).

It's also perfectly fine to start with really simple stuff and iterate on them later ([this page](https://reasonml.github.io/guide/javascript/converting) gives a good example on how you get more and more into Reason's type system).